editor/src/clj/editor/ui.clj

```diff
@@ -584,6 +584,15 @@
 (defn suppress-auto-commit! [^Node node]
   (user-data! node ::suppress-auto-commit true))
 
+(defn- clear-auto-commit! [^Node node]
+  ;; Clear the auto-commit flag. You should call this whenever data has been
+  ;; synced with the graph while the field still has focus. This ensures the
+  ;; unedited value will not be committed to the graph unnecessarily after the
+  ;; user moves focus to another control. Further edits will re-apply the
+  ;; auto-commit flag, but if they leave without editing, we won't commit.
+  (when (user-data node ::auto-commit)
+    (user-data! node ::auto-commit false)))
+
```

Comment:
Broke out this into a function and added longer explanation.

Reply:
yes, sure.